scripts/hooks/pre-commit.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+
+# check_golangci_lint_version ensures that the local version of golangci-lint
+# matches the version specified in .github/workflows/golangci.yml. This makes sure
+# that if the pre-commit hook is run locally, the changes should align with CI.
+function check_golangci_lint_version(){
+  local git_root="$(git rev-parse --show-toplevel)"
+
+  # extract the version of golangci-lint from the CI workflow file.
+  local workflow_golangci_lint_version="$(grep ' version' ${git_root}/.github/workflows/golangci.yml | awk '{ print $NF }')"
+
+  local local_golangci_lint_version="$(golangci-lint version --format short)"
+
+  if [[ "${workflow_golangci_lint_version}" != "${local_golangci_lint_version}" ]]; then
```

Comment:
testing this locally! Bumped into something, workflow version has a `v` in front of it, the comparison here will always fail, you need more bash.

Reply:
so looks like if you install with go get, it includes the v, but not with other versions 🤔 

```bash
~/go/bin/golangci-lint version --format short
v1.53.1
```